wled00/FX_2Dfcn.cpp

```diff
@@ -43,8 +43,17 @@ void WS2812FX::setUpMatrix() {
 
   // isMatrix is set in cfg.cpp or set.cpp
   if (isMatrix) {
-    Segment::maxWidth  = hPanels * panelW;
-    Segment::maxHeight = vPanels * panelH;
+    // calculate width dynamically because it will have gaps
+    matrixWidth = 0;
```

Comment:
You will need to revert to `Segment::maxWidth` and `Segment::maxHeight`.

Do similar for the code below.